adr/0005-code-formatting.md

```diff
@@ -0,0 +1,28 @@
+# 4. Code Formatting
+
+Date: 2019-07-07
+
+Architecture issue: [#249](https://github.com/home-assistant/architecture/issues/249)
+
+## Status
+
+Accepted
+
+## Context
+
+Proper code formatting is vital to the long-term viability and maintainability of a code
+base. In place, a codebase with such formatting is:
+
+- easier to review
+- easier to diff against previous commits
+- more likely to be free of linting errors
+
+## Proposal
+
+- We use [Black](https://github.com/python/black) as the de-facto code formatter for the entire back-end codebase
+- We use Black's default options
+- We utilize a bot (such as [the black_out bot](https://github.com/Mariatta/black_out)) to ensure PRs are appropriately formatted before being accepted
+
+## Consequences
+
+- We will need to determine what to do with style checking by `pylint`, `flake8`, etc.
```

Comment:
We should determine this before finalizing this proposal. Determining it shouldn't be part of consequences. Rather what changes to pylint and flake8 config could be part of consequences.

Reply:
What's the process for determining that? So far, there are two proposals:

* Mine: https://github.com/home-assistant/architecture/issues/249#issuecomment-499709271
* @frenck's: https://github.com/home-assistant/architecture/issues/249#issuecomment-499803823

Reply:
Make a suggestion on all rules to disable for pylint and flake8 etc. Then we come to a decision.

Reply:
I'm with Frenck, let's disable all formatting checks because we are no longer manually formatting things.